drivers/input/input_esp32_touch_sensor.c

```diff
@@ -0,0 +1,335 @@
+};
+
+struct touch_sensor_channel_data {
+	const struct device *dev;
+	struct k_work_delayable work;
+	uint32_t status;
+#if defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3)
+	uint32_t last_status;
+#endif /* defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3) */
+};
+
+struct touch_sensor_data {
+	uint32_t rtc_intr_msk;
+	struct touch_sensor_channel_data *channel_data;
+};
+
+static void touch_sensor_interrupt_cb(void *arg)
+{
+	const struct device *dev = (struct device *)arg;
+	struct touch_sensor_data *dev_data = dev->data;
+	const struct touch_sensor_config *dev_cfg = dev->config;
+	const struct touch_sensor_channel_config *channel_cfg;
+	const int num_channels = dev_cfg->num_channels;
+	uint32_t pad_status;
+
+#if defined(CONFIG_SOC_SERIES_ESP32)
+	touch_hal_intr_clear();
+
+#elif defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3)
+	static uint8_t scan_done_counter;
+
+	touch_pad_intr_mask_t intr_mask = touch_hal_read_intr_status_mask();
+
+	if (intr_mask & TOUCH_PAD_INTR_MASK_SCAN_DONE) {
+		if (scan_done_counter++ == SCAN_DONE_MAX_COUNT) {
+			touch_hal_intr_disable(TOUCH_PAD_INTR_MASK_SCAN_DONE);
+			for (int i = 0; i < num_channels; i++) {
+				channel_cfg = &dev_cfg->channel_cfg[i];
+
+				/* Set interrupt threshold */
+				uint32_t benchmark_value;
+
+				touch_hal_read_benchmark(channel_cfg->channel_num,
+					&benchmark_value);
+				touch_hal_set_threshold(channel_cfg->channel_num,
+				(channel_cfg->channel_sens * benchmark_value / 100));
+			}
+		}
+		return;
+	}
+#endif /* defined(CONFIG_SOC_SERIES_ESP32) */
+
+	touch_hal_read_trigger_status_mask(&pad_status);
+#if defined(CONFIG_SOC_SERIES_ESP32)
+	touch_hal_clear_trigger_status_mask();
+#endif /* defined(CONFIG_SOC_SERIES_ESP32) */
+	for (int i = 0; i < num_channels; i++) {
+		uint32_t channel_status;
+
+		channel_cfg = &dev_cfg->channel_cfg[i];
+		channel_status = (pad_status >> channel_cfg->channel_num) & 0x01;
+
+#if defined(CONFIG_SOC_SERIES_ESP32)
+		if (channel_status != 0) {
+#elif defined(CONFIG_SOC_SERIES_ESP32S2) || defined(CONFIG_SOC_SERIES_ESP32S3)
+		uint32_t channel_num = (uint32_t)touch_hal_get_current_meas_channel();
+
+		if (channel_cfg->channel_num == channel_num) {
+#endif /* CONFIG_SOC_SERIES_ESP32 */
+			struct touch_sensor_channel_data
+				*channel_data = &dev_data->channel_data[i];
+
+			channel_data->status = channel_status;
+			(void)k_work_reschedule(&channel_data->work,
+						K_MSEC(dev_cfg->debounce_interval_ms));
+		}
+	}
+}
+
+static void rtc_isr(void *arg)
+{
+	uint32_t status = REG_READ(RTC_CNTL_INT_ST_REG);
+
+	if (arg != NULL) {
+		const struct device *dev = (struct device *)arg;
+		struct touch_sensor_data *dev_data = dev->data;
+
+		if (dev_data->rtc_intr_msk & status) {
+			touch_sensor_interrupt_cb(arg);
+		}
+	}
+
+	REG_WRITE(RTC_CNTL_INT_CLR_REG, status);
+}
+
+static esp_err_t rtc_isr_install(intr_handler_t intr_handler, void *handler_arg)
+{
+	esp_err_t err = ESP_OK;
```

Comment:
drop the initializer, you set it directly just a couple lines below

Reply:
My bad